### PR TITLE
[6.x] Scripts/Spells: K'ure & Decimatus, SAI + spell scripts.

### LINKED ISSUE #15433: [6.x] Dark Syphon spell
https://github.com/TrinityCore/TrinityCore/commit/f4724badb65a6471df37ef249929548861f86362
hash.
Warlord of Draenor 6.x.x
http://www.wowhead.com/npc=81556/kure
http://www.wowhead.com/spell=165059/dark-siphon
``` SQL
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=13 AND `SourceGroup`=0 AND `SourceEntry`=165059;
INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
(13,1,165059,0,0,31,0,3,81556 ,0,0,"","Dark Siphon – Cast only on K'ure.");
```

### LINKED ISSUE #16175: [6.x] DB/Creature: Decimatus
https://github.com/TrinityCore/TrinityCore/commit/c08623e19a98890fea9e02a822c2d5720ae44dc1
hash.
6.x.x WoD
http://www.wowhead.com/npc=81989/decimatus#abilities
``` SQL
-- Decimatus SAI
SET @ENTRY := 81989;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,0,0,0,11,0,100,0,0,0,0,0,11,169869,34,0,0,0,0,1,0,0,0,0,0,0,0,"Decimatus - On Respawn - Cast Transformation Sickness on Self (Need Core Handler for Spell)"),
(@ENTRY,0,1,0,0,0,100,0,9000,11000,9000,11000,11,174011,0,0,0,0,0,5,0,0,0,0,0,0,0,"Decimatus - In Combat - Cast Neutralize - Need Core Handler."),
(@ENTRY,0,2,0,0,0,100,0,27000,27000,27000,27000,11,174016,1,0,0,0,0,1,0,0,0,0,0,0,0,"Decimatus - In Combat - Cast Decimate.");
```
For Areatrigger 3258 (MiscValueA for spell 173994) maybe need new SAI event, because SAI areatrigger on current state working only for static areatrigger from dbc.
http://www.wowhead.com/spell=173994/neutralize
SMART_EVENT_AREATRIGGER_ONTRIGGER only for static areatrigger working.
Timers from video and logs.

### PR DESCRIPTION
**Pull request by ghost**
**Changes proposed**:
- Add Decimatus SAI
- Add Dark Siphon K'ure spell condition target.
- Add Decimatus Transformation Sickness spell script.

**Target branch(es)**: 335/6x
6.x
**Issues addressed**: Fixes #
closes #16175
closes #15433
**Tests performed**: (Does it build, tested in-game, etc)
SAI and spell condition tested, spell script not tested.
Update:
Spell 169869 and 189491 testing in game, working, you can merge.
**Known issues and TODO list**:
- [x]  https://github.com/TrinityCore/TrinityCore/pull/17343#discussion-diff-66719297
  Maybe this method better, or all right?
- [x]  Spell not have basepoint, used number value.

``` CPP
ID - 169869 Transformation Sickness
=================================================

ToolTip: Decimatus has been temporarily weakened by the premature transformation.
Category = 0, SpellIconID = 6176, activeIconID = 0, SpellVisual = (42387,0)
Family SPELLFAMILY_GENERIC, flag [0] 0x00000000 [1] 0x00000000 [2] 0x00000000 [3] 0x00000000

SpellSchoolMask = 1 (SPELL_SCHOOL_MASK_NORMAL)
DamageClass = 0 (SPELL_DAMAGE_CLASS_NONE)
PreventionType = 0 (SPELL_PREVENTION_TYPE_NONE)
=================================================
Attributes: 0x00000100 (SPELL_ATTR0_HIDE_IN_COMBAT_LOG)
=================================================
Spell Level = 0, base 0, max 0

Category = 0
DispelType = 0 (DISPEL_NONE)
Mechanic = 0 (MECHANIC_NONE)
SpellRange: (Id 1) "Self Only":
    MinRangeNegative = 0, MinRangePositive = 0
    MaxRangeNegative = 0, MaxRangePositive = 0

Cast time (Id 1) = 0,00
Duration: ID (21)  -1, 0, -1
Interrupt Flags: 0x00000000, AuraIF 0x00000000, ChannelIF 0x00000000
Chance = 0, charges - 0
=================================================
Effect 0: Id 6 (SPELL_EFFECT_APPLY_AURA) DIFFICULTY_NONE
BasePoints = 0
Targets (1, 0) (TARGET_UNIT_CASTER, NO_TARGET)
Aura Id 4 (SPELL_AURA_DUMMY), value = 0, misc = 0 (0), miscB = 0, periodic = 0

Effect 1: Id 77 (SPELL_EFFECT_SCRIPT_EFFECT) DIFFICULTY_NONE
BasePoints = 0
Targets (1, 0) (TARGET_UNIT_CASTER, NO_TARGET)
```

**NOTE** If your Pull Request is SQL only create a ticket instead

**SUGESTION** If your Pull Request contains SQL give it one imposible date, for example 9999_99_99_99_database.sql on that way it will be free on merge.

### CODE REVIEW

**Comment by Keader** [comment]
Sql name.
Should conflict when merge.

**Comment by Aokromes** [comment]
Na, it won't, no one merged db content on 6.x branch that day.

File: src/server/scripts/Spells/spell_generic.cpp [hunk 1]
```diff
@@ -4228,6 +4228,34 @@ class spell_gen_clear_debuffs : public SpellScriptLoader
...
+
+    class spell_gen_decimatus_transformation_sickness_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_gen_decimatus_transformation_sickness_SpellScript);
+
+        void HandleScript(SpellEffIndex /*effIndex*/)
+        {
+            if (Unit* caster = GetCaster())
+                caster->SetHealth(caster->CountPctFromMaxHealth(10));
+        }
```
Comment (Ulduar) [inline src/server/scripts/Spells/spell_generic.cpp]: void HandleScript(SpellEffIndex /*effIndex*/)
         {
            if (Unit* caster = GetCaster())
                uint32 hp = caster->CountPctFromMaxHealth(10);
                if (hp)
                    caster->SetHealth(hp);
         }
Maybe this better? Also add in database HealthRegen interrupt flag.

**Comment by Ulduar** [comment]
``` QUOTE
Sql name.
Should conflict when merge.
```
Don't worry, I was staring sql date before add file.

File: src/server/scripts/Spells/spell_generic.cpp (hunk 1 above)
Comment (MitchesD) [inline src/server/scripts/Spells/spell_generic.cpp]: caster can't be null there, so useless check

Comment (Ulduar) [inline src/server/scripts/Spells/spell_generic.cpp]:
``` QUOTE
caster can't be null there, so useless check
```
Okay, you can suggested new variant? See below.
``` CPP
void HandleScript(SpellEffIndex /*effIndex*/)
        {
            if (Unit* target = GetHitUnit())
               target->SetHealth(target->CountPctFromMaxHealth(10));
        }
```
Right?

**Pull request merged**